### ISSUE TITLE
F# fails to determine which overload of `TaskSeqBuilder.Using` to use.

### ISSUE THREAD

**Issue by peterfaria-lula** [fsprojects/FSharp.Control.TaskSeq#97]
F# fails to determine which overload of `TaskSeqBuilder.Using` to use when type implements both `IDisposable` and `IAsyncDisposable`.

A real-world example with the [`NpgsqlDataReader`](https://github.com/npgsql/npgsql/blob/main/src/Npgsql/NpgsqlDataReader.cs) type from [Npgsql](https://github.com/npgsql/npgsql):

```F#
#r "nuget: Npgsql"
#r "nuget: FSharp.Control.TaskSeq"

open Npgsql
open FSharp.Control


let readRows (command: NpgsqlCommand) (ct: CancellationToken) f = taskSeq {
    use! reader = command.ExecuteReaderAsync ct // Fails to compile.
    
    let! reader = command.ExecuteReaderAsync ct
    use reader = reader // Also fails to compile.
}
```

Below is a set of test cases replicating the issue. 

```F#
#r "nuget: FSharp.Control.TaskSeq"
#r "nuget: Xunit"
#r "nuget: FsUnit"

open System
open System.Threading.Tasks
open FSharp.Control
open FsUnit
open Xunit


type private OneGetter() =
    member _.Get1() = 1

type private Disposable() =
    inherit OneGetter()

    interface IDisposable with
        member _.Dispose() = ()

type private AsyncDisposable() =
    inherit OneGetter()

    interface IAsyncDisposable with
        member _.DisposeAsync() = ValueTask()

type private MultiDispose() =
    inherit OneGetter()

    interface IDisposable with
        member _.Dispose() =
            ()

    interface IAsyncDisposable with
        member _.DisposeAsync() =
            ValueTask()

let private check ts = task {
    let! length = ts |> TaskSeq.length
    length |> should equal 1
}

[<Fact>]
let ``CE task: Using when type implements IDisposable``() =
    let ts = taskSeq {
        use x = new Disposable()

        yield x.Get1()
    }

    check ts

[<Fact>]
let ``CE task: Using when type implements IAsyncDisposable``() =
    let ts = taskSeq {
        use x = AsyncDisposable()
        yield x.Get1()
    }

    check ts


[<Fact>]
let ``CE task: Using when type implements IDisposable and IAsyncDisposable``() =
    let ts = taskSeq {
        use x = new MultiDispose() // Fails to compile
        yield x.Get1()
    }

    check ts

[<Fact>]
let ``CE task: Using! when type implements IDisposable``() =
    let ts = taskSeq {
        use! x = task { return new Disposable() }
        yield x.Get1()
    }

    check ts


[<Fact>]
let ``CE task: Using! when type implements IAsyncDisposable``() =
    let ts = taskSeq {
        use! x = task { return AsyncDisposable() }
        yield x.Get1()
    }

    check ts


[<Fact>]
let ``CE task: Using! when type implements IDisposable and IAsyncDisposable``() =
    let ts = taskSeq {
        use! x = task { return new MultiDispose() } // Fails to compile
        yield x.Get1()
    }

    check ts
```

**Comment by TheAngryByrd** [comment]
The `Using` implementation for `IDisposble` will need to be moved to a lower overload resolution.  See [IcedTask](https://github.com/TheAngryByrd/IcedTasks/blob/61e13a06c25683370bcf2bd91535084721f4005d/src/IcedTasks/CancellableTask.fs#L508-L513) implementation for an example.

**Comment by abelbraaksma** [comment]
Solved in #96, by adding priority overload modules similar to `tasks.fs` in F# Core. Will be part of release v0.3.0.
